### DISCUSSION TITLE
Thank you, and a question

### DISCUSSION THREAD

**Discussion by tr7zw** [gotson/NightMonkeys#4] [Q&A] [answered]
First of all, this is a really awesome project: using the new Java 18 Foreign Linker API(instead of JNA) and adding support for JPEG XL to the normal Java ImageIO, meaning using/adopting this lib is really easy to do. You might want to make a post on r/Java to get some more people looking into it.

To the question, are there any plans/roadblocks/ETA for adding writing xlr files? Looking at [komga](https://github.com/gotson/komga), it's probably the end goal to be able to also read in png files and send them to the client in jxr to save bandwidth on mobile devices.

**Comment by gotson** [comment]
> To the question, are there any plans/roadblocks/ETA for adding writing xlr files?

what is xlr exactly ? I have never heard of it, and a quick google search didn't return results.

> Looking at [komga](https://github.com/gotson/komga), it's probably the end goal to be able to also read in png files and send them to the client in jxr to save bandwidth on mobile devices.

That is not at all a goal for Komga, at least not in a foreseeable future. Conversion can be done on the fly if the client cannot read the original format. Converting on the fly while streaming is a time consuming job and would add significant lag to the streaming. You would also ideally need to not transcode from a lossy to lossless format, in order not to add artefacts.

Anyhow, this is a bit off-topic, and would be better suited for Komga's Discord or Github Discussions.

**Comment by tr7zw** [comment]
Oh sorry typo, I meant jxl(so Jpeg XL). Being able to also write out jxl files could open the door for a lot of neat tools. Also, this isn't about Komga, just saw that you are the dev of it(and assumed you are working on this lib to integrate it into there).

**Comment by gotson** [comment, marked as answer]
Jxl encoding is still being worked on actively in libjxl, the interfaces are not stable yet. I won't be working on integrating it in NightMonkeys until it becomes stable.

I don't have much time at the moment to work on this project, but the current focus is to make webp decoding compliant with imageio. The current version is very basic and works for simple use cases, but fails for lots of other tests. I would also like to add support for webp animations.
